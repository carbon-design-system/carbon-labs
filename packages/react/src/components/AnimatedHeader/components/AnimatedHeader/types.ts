/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import type { BaseTileProps } from '../../components/Tiles/BaseTile/BaseTile';

export type Tile = BaseTileProps;

export interface TileGroup {
  id: number;
  label: string;
  tiles: Tile[];
}

export interface AriaLabels {
  welcome?: string;
  description?: string;
  collapseButton?: string;
  expandButton?: string;
  tilesContainer?: string;
}
