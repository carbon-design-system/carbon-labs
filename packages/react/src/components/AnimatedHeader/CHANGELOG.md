# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.16.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-animated-header@0.15.0...@carbon-labs/react-animated-header@0.16.0) (2025-06-26)

**Note:** Version bump only for package @carbon-labs/react-animated-header





# [0.15.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-animated-header@0.14.0...@carbon-labs/react-animated-header@0.15.0) (2025-06-18)


### Features

* add-tile-click-prop ([#663](https://github.com/carbon-design-system/carbon-labs/issues/663)) ([a3b1b46](https://github.com/carbon-design-system/carbon-labs/commit/a3b1b4662968f81581357965e584180483708d89))





# [0.14.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-animated-header@0.13.0...@carbon-labs/react-animated-header@0.14.0) (2025-05-28)


### Bug Fixes

* update exports to match UI shell ([#605](https://github.com/carbon-design-system/carbon-labs/issues/605)) ([1bf4682](https://github.com/carbon-design-system/carbon-labs/commit/1bf46822620b44cc1ad9ce58913bf26e9c3a2ca1))





# [0.13.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-animated-header@0.12.0...@carbon-labs/react-animated-header@0.13.0) (2025-05-08)


### Bug Fixes

* correct package exports ([#539](https://github.com/carbon-design-system/carbon-labs/issues/539)) ([ef7557e](https://github.com/carbon-design-system/carbon-labs/commit/ef7557e10c1385a6bdc60d32361ce7ba2dad263c))





# [0.12.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-animated-header@0.11.0...@carbon-labs/react-animated-header@0.12.0) (2025-04-29)

**Note:** Version bump only for package @carbon-labs/react-animated-header





# [0.11.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-animated-header@0.10.0...@carbon-labs/react-animated-header@0.11.0) (2025-04-18)

**Note:** Version bump only for package @carbon-labs/react-animated-header





# [0.10.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-animated-header@0.9.0...@carbon-labs/react-animated-header@0.10.0) (2025-04-10)


### Bug Fixes

* **animated-header:** add backup for autoprefixer ([#563](https://github.com/carbon-design-system/carbon-labs/issues/563)) ([8c3c4a6](https://github.com/carbon-design-system/carbon-labs/commit/8c3c4a6846627b6fbb8e65e50bb731acb52784dc))





# [0.9.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-animated-header@0.8.0...@carbon-labs/react-animated-header@0.9.0) (2025-04-10)


### Bug Fixes

* **animated-header:** cleanup Safari issues ([#556](https://github.com/carbon-design-system/carbon-labs/issues/556)) ([530e01a](https://github.com/carbon-design-system/carbon-labs/commit/530e01ae13d94391732bf34a3b26edb87b9ea161))





# [0.8.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-animated-header@0.7.0...@carbon-labs/react-animated-header@0.8.0) (2025-04-08)


### Features

* **animated-header:** remove strictmode from example app + tooltip ([#548](https://github.com/carbon-design-system/carbon-labs/issues/548)) ([c969903](https://github.com/carbon-design-system/carbon-labs/commit/c969903eab1bd7b78fbb3b7f9770e4839b1d7f6b))





# [0.7.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-animated-header@0.6.0...@carbon-labs/react-animated-header@0.7.0) (2025-04-04)

**Note:** Version bump only for package @carbon-labs/react-animated-header





# [0.6.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-animated-header@0.5.0...@carbon-labs/react-animated-header@0.6.0) (2025-04-01)


### Features

* **animated-header:** remove loop animation from header ([#547](https://github.com/carbon-design-system/carbon-labs/issues/547)) ([613ffc3](https://github.com/carbon-design-system/carbon-labs/commit/613ffc358e22b8270dc0535cd6d5d4a352dc8ac8))





# [0.5.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-animated-header@0.4.0...@carbon-labs/react-animated-header@0.5.0) (2025-04-01)


### Features

* **animated-header:** add custom itemToString func for Dropdowns ([#542](https://github.com/carbon-design-system/carbon-labs/issues/542)) ([8ae02cf](https://github.com/carbon-design-system/carbon-labs/commit/8ae02cf012adec82e3fd962bf3f84df0752023e9))





# [0.4.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-animated-header@0.3.0...@carbon-labs/react-animated-header@0.4.0) (2025-03-26)


### Bug Fixes

* **animated-header:** bug fixes ([#523](https://github.com/carbon-design-system/carbon-labs/issues/523)) ([35781f5](https://github.com/carbon-design-system/carbon-labs/commit/35781f5eb3371d38633f2ba9fbb28e4e25adade4))





# [0.3.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-animated-header@0.2.0...@carbon-labs/react-animated-header@0.3.0) (2025-03-04)


### Bug Fixes

* **animated-header:** fix dropdown bugs and rename tile-dropdown toâ€¦ ([#485](https://github.com/carbon-design-system/carbon-labs/issues/485)) ([a12c104](https://github.com/carbon-design-system/carbon-labs/commit/a12c104eb0fb0e40449a55af3013855dc539f271))





# [0.2.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-animated-header@0.1.0...@carbon-labs/react-animated-header@0.2.0) (2025-02-19)

**Note:** Version bump only for package @carbon-labs/react-animated-header





# 0.1.1 (2025-02-14)


### Features

* **animated-header:** Fix missing assets


# 0.1.0 (2025-02-14)


### Features

* **animated-header:** add new animated header component ([#406](https://github.com/carbon-design-system/carbon-labs/issues/406)) ([01832e9](https://github.com/carbon-design-system/carbon-labs/commit/01832e97d083f5680d37ddaf7a337df83eebed54))
