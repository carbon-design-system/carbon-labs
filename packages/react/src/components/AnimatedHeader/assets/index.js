/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import dataFabricAnimatedLight from './animated/header_1312x738_data_fabric_light_03.json';
import dataFabricAnimatedDark from './animated/header_1312x738_data_fabric_dark_06.json';
import dataFabricStaticLight from './static/illustration_data_fabric_light_04.svg';
import dataFabricStaticDark from './static/illustration_data_fabric_dark_05.svg';
import watsonXAnimatedLight from './animated/header_1312x738_watsonx_data_light_03.json';
import watsonXAnimatedDark from './animated/header_1312x738_watsonx_data_dark_02.json';
import watsonXStaticLight from './static/illustration_watson_x_data_light_03.svg';
import watsonXStaticDark from './static/illustration_watson_x_data_dark_03.svg';
import wxbiaAnimatedLight from './animated/header_illustration_container_wxbia_light_08.json';
import wxbiaAnimatedDark from './animated/header_illustration_container_wxbia_dark_05.json';

export {
  dataFabricAnimatedLight,
  dataFabricAnimatedDark,
  dataFabricStaticLight,
  dataFabricStaticDark,
  watsonXAnimatedLight,
  watsonXAnimatedDark,
  watsonXStaticLight,
  watsonXStaticDark,
  wxbiaAnimatedLight,
  wxbiaAnimatedDark,
};
