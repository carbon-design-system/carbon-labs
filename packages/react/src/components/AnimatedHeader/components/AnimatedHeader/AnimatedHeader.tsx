/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import PropTypes from 'prop-types';
import React, {
  ElementType,
  ReactNode,
  useEffect,
  useRef,
  useState,
} from 'react';
import { Grid, Column, Button, Tooltip } from '@carbon/react';
import { ChevronUp, ChevronDown } from '@carbon/icons-react';
import lottie from 'lottie-web';
import { usePrefix } from '@carbon-labs/utilities/es/index.js';

import { BaseTile } from '../Tiles/index';
import TasksController, {
  TasksControllerProps,
} from '../TasksController/TasksController';
import WorkspaceSelector, {
  WorkspaceSelectorProps,
} from '../WorkspaceSelector/WorkspaceSelector';

/** Animated Header */

export interface Tile {
  href?: string | null;
  id: string;
  mainIcon?: ElementType | null;
  secondaryIcon?: ElementType | null;
  subtitle?: string | null;
  title?: string | null;
  customContent?: ReactNode | null;
  isLoading?: boolean;
  isDisabled?: boolean;
  isInteractive?: boolean;
}

export interface TileGroup {
  id: number;
  label: string;
  tiles: Tile[];
}

export type AnimatedHeaderProps = {
  allTileGroups?: TileGroup[];
  selectedTileGroup?: TileGroup;
  setSelectedTileGroup: (e) => void;
  description?: string;
  headerAnimation?: object;
  headerStatic?: React.JSX.Element;
  productName?: string;
  userName?: string;
  welcomeText?: string;
  isLoading?: boolean;
  disabledTaskLabel?: string;
  expandButtonLabel?: string;
  collapseButtonLabel?: string;
  tileClickHandler?: (tile: Tile) => void;
} & TasksControllerProps &
  WorkspaceSelectorProps;

const AnimatedHeader: React.FC<AnimatedHeaderProps> = ({
  allTileGroups,
  selectedTileGroup,
  setSelectedTileGroup,
  description,
  headerAnimation,
  headerStatic,
  productName = '[Product name]',
  userName,
  welcomeText,
  tasksControllerConfig,
  workspaceSelectorConfig,
  isLoading,
  disabledTaskLabel,
  expandButtonLabel = 'Expand',
  collapseButtonLabel = 'Collapse',
  tileClickHandler,
}: AnimatedHeaderProps) => {
  const prefix = usePrefix();
  const blockClass = `${prefix}--animated-header`;

  const animationContainer = useRef<HTMLDivElement>(null);
  const [open, setOpen] = useState(true);
  const [headingTextAnimation, setHeadingTextAnimation] = useState('');
  const isReduced = window.matchMedia('(prefers-reduced-motion)').matches;

  const collapsed = `${blockClass}--collapsed`;
  const contentCollapsed = `${blockClass}__content--collapsed`;
  const descriptionCollapsed = `${blockClass}__left-area-container--collapsed`;
  const headingCollapsed = `${blockClass}__heading--collapsed`;
  const headingExpanded = `${blockClass}__heading--expanded`;
  const lottieCollapsed = `${blockClass}__lottie-animation--collapsed`;

  const handleButtonCollapseClick = () => {
    setOpen(!open);

    open
      ? setHeadingTextAnimation(headingCollapsed)
      : setHeadingTextAnimation(headingExpanded);
  };

  useEffect(() => {
    if (!animationContainer.current) {
      return;
    }

    const animation = lottie.loadAnimation({
      container: animationContainer.current as HTMLDivElement,
      animationData: headerAnimation,
      renderer: 'svg',
      loop: false,
      autoplay: false,
    });

    const lottieLoadSpeed = 1;
    const animationData = animation['animationData'];

    function reducedMotion() {
      animation.goToAndStop(5000);
    }

    function load() {
      animation.setSpeed(lottieLoadSpeed);
      animation.playSegments(
        [animationData.markers.at(0).tm, animationData.markers.at(1).tm],
        true
      );
    }

    if (isReduced) {
      // reduced motion on
      animation.addEventListener('DOMLoaded', reducedMotion);
    } else {
      // reduced motion off

      // Run Start Animation
      animation.addEventListener('DOMLoaded', load);
    }
    return () => {
      animation?.removeEventListener('DOMLoaded', reducedMotion);
      animation?.removeEventListener('DOMLoaded', load);
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isReduced]);

  return (
    <section className={`${blockClass}${!open ? ` ${collapsed}` : ''}`}>
      <Grid>
        <div className={`${blockClass}__gradient--overlay`} />

        <div className={`${blockClass}__container--gradient`} />

        {!headerAnimation && headerStatic && (
          <div className={`${blockClass}__static--container`}>
            <div
              className={`${blockClass}__static`}
              // eslint-disable-next-line react/forbid-dom-props
              style={{ backgroundImage: `url(${headerStatic})` }}
            />
          </div>
        )}

        <div className={`${blockClass}__lottie-animation--container`}>
          <div
            ref={animationContainer}
            className={`${blockClass}__lottie-animation${
              !open ? ` ${lottieCollapsed}` : ''
            }`}></div>
        </div>

        <Column sm={4} md={8} lg={16}>
          <Tooltip align="bottom" label={`${welcomeText}, ${userName}`}>
            <h1 className={`${blockClass}__heading ${headingTextAnimation}`}>
              <span className={`${blockClass}__heading-welcome`}>
                {welcomeText},{' '}
              </span>
              <span className={`${blockClass}__heading-name`}>{userName}</span>
            </h1>
          </Tooltip>
        </Column>

        {(description || tasksControllerConfig) && (
          <Column
            sm={4}
            md={8}
            lg={4}
            className={`${blockClass}__left-area-container${
              !open ? ` ${descriptionCollapsed}` : ''
            }`}>
            {description && (
              <h2 className={`${blockClass}__description`}>{description}</h2>
            )}
            <TasksController
              tasksControllerConfig={tasksControllerConfig}
              isLoading={isLoading}
              allTileGroups={allTileGroups}
              selectedTileGroup={selectedTileGroup}
              setSelectedTileGroup={setSelectedTileGroup}
            />
          </Column>
        )}

        {selectedTileGroup && (
          <Column sm={4} md={8} lg={12} className={`${blockClass}__content`}>
            {workspaceSelectorConfig?.allWorkspaces?.length && (
              <div
                className={`${blockClass}__workspace--container${
                  !open ? ` ${contentCollapsed}` : ''
                }`}>
                <WorkspaceSelector
                  workspaceSelectorConfig={workspaceSelectorConfig}
                  userName={userName}
                  isLoading={isLoading}
                />
              </div>
            )}
            <div className={`${blockClass}__tiles-container`}>
              {selectedTileGroup.tiles.map((tile) => {
                return (
                  <BaseTile
                    onClick={() => tileClickHandler?.(tile)}
                    key={tile.id}
                    id={tile.id}
                    open={open}
                    href={tile.href}
                    mainIcon={tile.mainIcon}
                    secondaryIcon={tile.secondaryIcon}
                    title={tile.title}
                    subtitle={tile.subtitle}
                    productName={productName}
                    customContent={tile.customContent}
                    isLoading={isLoading || tile.isLoading}
                    isDisabled={tile.isDisabled}
                    disabledTaskLabel={disabledTaskLabel}
                    isInteractive={tile.isInteractive}
                  />
                );
              })}
            </div>
          </Column>
        )}

        <div className={`${blockClass}__button-collapse--gradient`} />

        <div className={`${blockClass}__button-collapse--container`}>
          <Button
            id={`${blockClass}__button-collapse`}
            kind="ghost"
            renderIcon={open ? ChevronUp : ChevronDown}
            onClick={handleButtonCollapseClick}>
            {open ? collapseButtonLabel : expandButtonLabel}
          </Button>
        </div>
      </Grid>
    </section>
  );
};

(AnimatedHeader as React.FC).displayName = 'Animated Header';
(AnimatedHeader as React.FC).propTypes = {
  /**
   * Array of each tile group setup
   */
  allTileGroups: PropTypes.arrayOf(PropTypes.object),

  /**
   * Specify an optional className to be added to your Animated Header
   */
  className: PropTypes.string,

  /**
   * Custom collapse button label
   */
  collapseButtonLabel: PropTypes.string,

  /**
   * Provide short sentence in max. 3 lines related to product context
   */
  description: PropTypes.string,

  /**
   * Custom expand button label
   */
  expandButtonLabel: PropTypes.string,

  /**
   * In-product imagery / lottie animation (.json) dim. 1312 x 738
   * (to update headerAnimation content storybook requires remount in toolbar)
   */
  headerAnimation: PropTypes.object,

  /**
   * In-product imagery / static imagery dim. 1312 x 738
   * Only active when headerAnimation is not in use
   */
  headerStatic: PropTypes.object,

  /**
   * Check if is loading
   */
  isLoading: PropTypes.bool,

  /**
   * Provide current product name
   */
  productName: PropTypes.string,

  /**
   * The tile group that is active in the header
   * ex. "AI Chat Tile w/ two glass tiles", "Four glass tiles", ect.
   */
  selectedTileGroup: PropTypes.object,

  /**
   * Provide function to be called when switching selected tile group
   */
  setSelectedTileGroup: PropTypes.func,

  /**
   * Configuration for Carbon button or dropdown menu in header. Customized
   * tasks are used to allow users that have multiple roles and permissions
   * to experience better tailored content based on their need.
   * It also allows to override Carbon Button props by specifying them in tasksConfig.button.propsOverrides
   * or to override Carbon Dropdown props by specifying them in tasksConfig.dropdown.propsOverrides.
   */
  tasksControllerConfig: PropTypes.shape({
    type: PropTypes.oneOf(['button', 'dropdown']).isRequired,
    button: PropTypes.shape({
      text: PropTypes.string.isRequired,
      // Override Carbon Button props
      propsOverrides: PropTypes.object,
    }),
    dropdown: PropTypes.shape({
      allTileGroups: PropTypes.arrayOf(PropTypes.object),
      selectedTileGroup: PropTypes.object,
      setSelectedTileGroup: PropTypes.func.isRequired,
      // Override Carbon Dropdown props
      propsOverrides: PropTypes.object,
    }),
  }),

  /**
   * Handler for tile clicks
   */
  tileClickHandler: PropTypes.func,

  /**
   * Specify the current username of active user
   */
  userName: PropTypes.string,

  /**
   * Specify the current welcome text on the header
   */
  welcomeText: PropTypes.string,

  /**
   * Configuration for Carbon dropdown for workspace selection. To override Carbon Dropdown props use workspaceSelectorConfig.propsOverride
   */
  workspaceSelectorConfig: PropTypes.shape({
    // Override Carbon Dropdown props
    propsOverrides: PropTypes.object,
    allWorkspaces: PropTypes.arrayOf(PropTypes.object),
    selectedWorkspace: PropTypes.object,
    setSelectedWorkspace: PropTypes.func.isRequired,
  }),
};

export default AnimatedHeader;
