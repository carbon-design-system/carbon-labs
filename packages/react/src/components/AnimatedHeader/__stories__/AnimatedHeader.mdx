import { ArgTypes, Canvas, Meta } from '@storybook/blocks';
import * as AnimatedHeaderStories from './AnimatedHeader.stories';
import AnimatedHeader from '../components/AnimatedHeader/AnimatedHeader';

<Meta isTemplate />

# Animated Header

- **Initiative owner(s):** Drew Glapa
- **Status:** Draft
- **Target library:** TBD
- **Target library maintainer(s) / PR Reviewer(s):** N/A
- **Support channel:** `#carbon-labs`

{/* <!-- START doctoc generated TOC please keep comment here to allow auto update --> */}
{/* <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE --> */}

> 💡 Check our
> [Stackblitz](https://stackblitz.com/github/carbon-design-system/carbon-labs/tree/main/examples/react/AnimatedHeader)
> example implementation.

[![Edit carbon-labs](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/github/carbon-design-system/carbon-labs/tree/main/examples/react/AnimatedHeader)

## Table of Contents

- [Overview](#overview)
- [Getting started](#getting-started)

{/* <!-- END doctoc generated TOC please keep comment here to allow auto update --> */}

## Overview

The animated header is meant to be used on homepages only. It ensures
consistency throughout landing experiences, providing a cohesive appearance and
clear call to action for users. The skeleton for this header component ensures
consistency while flexible page elements allow teams to tailor the leadspace to
their individual user and product needs.

Lead spaces include a welcome text, a short description communicating the
products value proposition and feature most relevant content for the user.
Whether these are links to specific tasks, workspaces or other information is
defined by each product and their user’s needs. In product-imagery has to be
very settled and subtil to not distract users in performing their tasks.

<Canvas
  of={AnimatedHeaderStories.ThemeG10}
  additionalActions={[
    {
      title: 'Open in Stackblitz',
      onClick: () => stackblitzPrefillConfig(AnimatedHeaderStories.ThemeG10),
    },
  ]}
/>

### Anatomy

**Title:** Welcomes the user (Welcome, username)

**Content:** Short sentence in max. 3 lines related to product context; Action
button (optional), Expose content that need highest attention from the user or
content that trigger an action and allow users to directly start working and
gain value (within one click). It is recommended to display content on cards to
achieve highest contrast to the background.

**Visual / brand image:** Line-style illustration

**Collapse / Expand:** Icon to collapse or expand the header

### Theming

The default theme for Animated Header is _g10_ (Light theme).

Animated header supports both _g10_ (Light theme) and _g100_ (Dark Theme).

To use dark theme in your product/app wrap the componenet in the Carbon
[GlobalTheme](https://react.carbondesignsystem.com/?path=/docs/components-theme--overview#setting-the-global-theme)
component. Following the example from the Carbon website it would like this:

```
import { useEffect } from 'react';
import { GlobalTheme } from '@carbon/react';

function App() {
  const theme = 'g100'; // ← your implementation, e.g. fetching user settings

  useEffect(() => {
    document.documentElement.dataset.carbonTheme = theme;
  }, [theme]);

  return (
    <GlobalTheme theme={theme}>
      <AnimatedHeader />
    </GlobalTheme>;
  );
}
```

## Getting started

Here's a quick example to get you started.

```bash
yarn add @carbon/react
yarn add @carbon-labs/react-animated-header
```

### JS (via import)

```javascript
import { AnimatedHeader } from '@carbon-labs/react-animated-header/es/index';

function App() {
  return <AnimatedHeader />;
}

export default App;
```

### SCSS

In your styles file import

```
@use '@carbon/react';
@use '@carbon-labs/react-animated-header/scss/animated-header';
```

## Component API

<ArgTypes />

## Example usages

Example of Animated Header with all prop types enabled.

```jsx
  <AnimatedHeader
    tasksControllerConfig={tasksControllerConfigDropdown as TasksControllerConfig}
    workspaceSelectorConfig={WorkspaceSelectorConfig}
    description="Connect, monitor, and manage data."
    productName="[Product name]"
    userName="Drew"
    welcomeText="Welcome"
    isLoading={false}
    expandButtonLabel="Expand"
    collapseButtonLabel="Collapse"
    tileClickHandler={handleTileClicks}
  />
```
