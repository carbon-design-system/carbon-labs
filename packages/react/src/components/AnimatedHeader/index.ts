/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import AnimatedHeader, {
  AriaLabels,
  TileGroup,
} from './components/AnimatedHeader/AnimatedHeader';
import HeaderTitle from './components/HeaderTitle/HeaderTitle';
import { BaseTile } from './components/Tiles/index.js';
export * from './assets';
export type {
  Workspace,
  WorkspaceSelectorConfig,
} from './components/WorkspaceSelector/WorkspaceSelector';
export type { TasksControllerConfig } from './components/TasksController/TasksController';

export {
  AnimatedHeader,
  BaseTile,
  HeaderTitle,
  type AriaLabels,
  type TileGroup,
};
