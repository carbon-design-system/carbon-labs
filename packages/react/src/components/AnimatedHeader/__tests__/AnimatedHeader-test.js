/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from 'react';
import { render } from '@testing-library/react';
import 'jest-canvas-mock';
import '@testing-library/jest-dom';

import AnimatedHeader from '../components/AnimatedHeader/AnimatedHeader';

window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      addListener: function () {},
      removeListener: function () {},
    };
  };

jest.mock('./animated-header.scss', () => ({}));
describe('AnimatedHeader', () => {
  describe('renders as expected - Component API', () => {
    it('should match snapshot', () => {
      const { container } = render(<AnimatedHeader />);
      expect(container).toMatchSnapshot();
    });
  });
});
