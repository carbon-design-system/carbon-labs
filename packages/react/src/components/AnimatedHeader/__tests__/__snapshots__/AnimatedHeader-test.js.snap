// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedHeader renders as expected - Component API should match snapshot 1`] = `
<div>
  <section
    class="clabs-animated-header false"
  >
    <div
      class="cds--css-grid"
    >
      <div
        class="clabs-animated-header__gradient--overlay"
      />
      <div
        class="clabs-animated-header__container--gradient"
      />
      <div
        class="clabs-animated-header__lottie-animation--container"
      >
        <div
          class="clabs-animated-header__lottie-animation false"
        />
      </div>
      <div
        class="cds--sm:col-span-4 cds--md:col-span-8 cds--lg:col-span-16 cds--css-grid-column"
      >
        <span
          class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--bottom cds--tooltip"
        >
          <div
            class="cds--tooltip-trigger__wrapper"
          >
            <h1
              aria-labelledby="tooltip-:r0:"
              class="clabs-animated-header__heading "
            >
              <span
                class="clabs-animated-header__heading-welcome"
              >
                Welcome,
                 
              </span>
              <span
                class="clabs-animated-header__heading-name"
              />
            </h1>
          </div>
          <span
            aria-hidden="true"
            class="cds--popover"
            id="tooltip-:r0:"
            role="tooltip"
          >
            <span
              class="cds--popover-content cds--tooltip-content"
            >
              Welcome, undefined
            </span>
            <span
              class="cds--popover-caret"
            />
          </span>
        </span>
      </div>
      <div
        class="clabs-animated-header__button-collapse--gradient"
      />
      <div
        class="clabs-animated-header__button-collapse--container"
      >
        <button
          class="cds--btn cds--btn--ghost"
          id="clabs-animated-header__button-collapse"
          type="button"
        >
          Collapse
          <svg
            aria-hidden="true"
            class="cds--btn__icon"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 5L13 10 12.3 10.7 8 6.4 3.7 10.7 3 10z"
            />
          </svg>
        </button>
      </div>
    </div>
  </section>
</div>
`;
